src/coreclr/tools/Common/InstructionSetHelpers.cs

```diff
@@ -34,7 +38,24 @@ internal static partial class Helpers
                 }
             }
 
-            if (instructionSet != null)
+            if (instructionSet == "native")
+            {
+                if (GetTargetArchitecture(null) != targetArchitecture)
+                {
+                    throw new CommandLineException("Instruction set 'native' not supported when cross-compiling to a different architecture.");
+                }
+
+                string jitInterfaceLibrary = "jitinterface_" + RuntimeInformation.ProcessArchitecture.ToString().ToLowerInvariant();
+                nint libHandle = NativeLibrary.Load(jitInterfaceLibrary, System.Reflection.Assembly.GetExecutingAssembly(), DllImportSearchPath.ApplicationDirectory);
+                int cpuFeatures;
+                unsafe
+                {
+                    var getCpuFeatures = (delegate* unmanaged<int>)NativeLibrary.GetExport(libHandle, "JitGetProcessorFeatures");
+                    cpuFeatures = getCpuFeatures();
+                }
+                HardwareIntrinsicHelpers.AddRuntimeRequiredIsaFlagsToBuilder(instructionSetSupportBuilder, cpuFeatures);
+            }
+            else if (instructionSet != null)
```

Comment:
Open issue: should we allow computing optimistic instruction set or skip that when `native` was specified.

(Or should we actually skip the optimistic one if _any_ `instruction-set` is specified?)

Reply:
Should we make it configurable? For example, we can introduce `optimistic` instruction set that would enable the optimistic checks; or we can introduce `?` qualifier for instruction sets (in addition to the existing `+` and '-` qualifiers) that would enable optimistic checks for given instruction set.

Reply:
That sounds like a good plan.

Reply:
Does that mean that you'd have to specify eg. `native,foo?,bar?,baz?` (with `foo` etc. being all opportunistic instruction sets you want guarded paths to be generated for) to get the resulting binary to use all the instruction sets available on the current machine, _and_ guarded paths for all the other ones? I wonder if it would make sense to have a simplified way to express:

> "Compile for `native`, but also add guarded paths for all existing other ISAs supported by the runtime".

Basically as a way to avoid having to manually look all of them up and add them with `?` as a suffix for each after `native`. Just to get the idea across, something like `native,all?`. Or is something like this already possible in some other way?